### PR TITLE
Adding activeWeapon to Player ESP with Config

### PR DESCRIPTION
**Pull request by 0xE232FE**
Hello Daniel,

Thank you for adding WeaponData!!!

I added activeWeapon to Player ESP. I find it usefull.

Maybe you take a look please?

The Positioning is not final, because I have to test it out. But you know better than me where it can be positioned.

Thank you very much!

best regards

### CODE REVIEW

**Comment by effex1337** [comment]
Good job

**Comment by 0xE232FE** [comment]
Additionally here is the Ammo ESP for Weapon ESP on the ground
`renderPositionedText(config.font, (std::wostringstream{ } << std::fixed << std::showpoint << std::setprecision(2) << (entity->clip()) << L'|' << (entity->getWeaponData()->maxClip)).str().c_str(), config.distanceColor, { bbox.right + 5, drawPositionY });`

**Comment by 0xE232FE** [comment]
I have a little bit trouble with string replacement. Any weapon contains the string from CSGO ClientClass `#SFUI_WPNHUD_` which is annoying. I remove the first 13 Chars from the name of the WeaponData. If someone have a better method for this. Please let me know.

**Comment by effex1337** [comment]
https://cdn.otaku.ooo/a7pqr.png
heres my weapon esp the font is a bit fked but it does the job
current ammo, maxclip, reserved ammo 😉

**Comment by anonymouschicken111** [comment]
Nice job!

**Comment by Bywnb** [comment]
Good!Daniel should pull this. @danielkrupinski

**Comment by moonm00n** [comment]
Is it only crashing for me?

File: Osiris/Hacks/Esp.cpp [hunk 1]
```diff
@@ -221,6 +221,32 @@ static void renderPlayerBox(Entity* entity, const Config::Esp::Player& config) n
             }
         }
 
+		if (config.activeWeapon)
+		{
+			static PlayerInfo playerInfo;
+			if (interfaces.engine->getPlayerInfo(entity->index(), playerInfo))
```
Comment (danielkrupinski) [inline Osiris/Hacks/Esp.cpp]: Unneeded

File: Osiris/Hacks/Esp.cpp [hunk 2]
```diff
@@ -221,6 +221,32 @@ static void renderPlayerBox(Entity* entity, const Config::Esp::Player& config) n
...
+			{
+				static wchar_t name[128];
+				if (MultiByteToWideChar(CP_UTF8, 0, entity->getActiveWeapon()->getWeaponData()->name, -1, name, 128))
```
Comment (danielkrupinski) [inline Osiris/Hacks/Esp.cpp]: Use https://github.com/danielkrupinski/Osiris/blob/71c1846eb7bdceda82b66745b0795c9fc9f480c6/Osiris/SDK/Localize.h#L7

File: Osiris/Config.cpp [hunk 3]
```diff
@@ -226,6 +226,12 @@ void Config::load(size_t id) noexcept
             espConfig.distanceColor[1] = espJson["Distance color"][1].asFloat();
             espConfig.distanceColor[2] = espJson["Distance color"][2].asFloat();
         }
+		if (espJson.isMember("activeWeapon")) espConfig.distance = espJson["activeWeapon"].asBool();
```
Comment (danielkrupinski) [inline Osiris/Config.cpp]: JSON key names should start with capital letter see above.

File: Osiris/Hacks/Esp.cpp (hunk 2 above)
Comment (0xE232FE) [reply Osiris/Hacks/Esp.cpp]: I use the Localize find token and its working!!! Thank you very much for the hint!

File: Osiris/Config.cpp (hunk 3 above)
Comment (0xE232FE) [reply Osiris/Config.cpp]: okay I will keep this in mind. I thaught Capital in Json means that the Variable is explicit Exposed. Isn't it?

**Comment by xsincbt** [comment]
Nice job I like it

**Pull request merged**